### PR TITLE
:warning: Exclude package labels, refactor lspServerPath, add language label

### LINKED ISSUE #198: Consider moving lspServerPath to providerSpecificConfig
Link to relevant [slack conversation](https://kubernetes.slack.com/archives/C04QZJFQ0UA/p1686167624905009?thread_ts=1686159109.450979&cid=C04QZJFQ0UA).

Main idea is that it may be better to let each provider handle the path (and how to invoke) the language server, and if a provider decides they want to allow the path to be overridden, they can handle that too.

### LINKED ISSUE #247: Add a provider specific config to allow passing a include / exclude pattern filters to work with dep label selector

### LINKED ISSUE #248: Probably need to remove Type, as that functionality is handled via labels
https://github.com/konveyor/analyzer-lsp/blob/2fee76eb8e4f7bdc7a12546873f113395503f56c/output/v1/konveyor/violations.go#L99C4-L99C4

### LINKED ISSUE #255: Add language label to dependencies
We need to add a label that tells which language it is for. We were hoping to use provider names but provider names are not accurate representation of the language.

### PR DESCRIPTION
**Pull request by pranavgaikwad**
Fixes #198 
Fixes #247  
Fixes #255
Additionally fixes the `downloadSources` setting I broke earlier :)

### CODE REVIEW

File: provider/internal/java/dependency.go [hunk 1]
```diff
@@ -261,21 +268,24 @@ func (p *javaServiceClient) parseDepString(dep, localRepoPath string) (provider.
...
 	}
 	s := []string{}
 	for k, _ := range m {
 		s = append(s, k)
 	}
-	if len(s) == 0 {
-		s = append(s, fmt.Sprintf("%v=%v", provider.DepSourceLabel, javaDepSourceInternal))
+	// if open source label is not found, qualify the dep as being internal by default
+	if _, openSourceLabelFound :=
+		m[fmt.Sprintf("%s=%s", provider.DepSourceLabel, javaDepSourceOpenSource)]; !openSourceLabelFound {
```
Comment (shawn-hurley) [inline provider/internal/java/dependency.go:284]: nit: in the future, should we consider making a label key helper?

File: provider/provider.go [hunk 2]
```diff
@@ -425,6 +417,28 @@ func (p *ProviderCondition) Evaluate(ctx context.Context, log logr.Logger, condC
 
 }
 
+// matchDepLabelSelector evaluates the dep label selector on incident
+func matchDepLabelSelector(s *labels.LabelSelector[*Dep], inc IncidentContext, deps map[uri.URI][]*konveyor.Dep) (bool, error) {
```
Comment (shawn-hurley) [inline provider/provider.go:421]: +1 on this refactor thank you

**Review by shawn-hurley — APPROVED** [review]
This looks good for now, and I like the refactoring, wondering if we want a generic way to add any label based on some config?

Maybe post Demo E?

File: external-providers/golang-external-provider/pkg/golang/provider.go [hunk 3]
```diff
@@ -51,9 +51,14 @@ func (p *golangProvider) Init(ctx context.Context, log logr.Logger, c provider.I
 		os.Setenv(k, v)
 	}
 
+	lspServerPath, ok := c.ProviderSpecificConfig[provider.LspServerPathConfigKey].(string)
+	if !ok || lspServerPath == "" {
+		return nil, fmt.Errorf("invalid lspServerPath provided, unable to init go provider")
```
Comment (fabianvf) [inline external-providers/golang-external-provider/pkg/golang/provider.go:56]: just curious, does this error bubble to the user nicely?

**Review by fabianvf — APPROVED** [review]

File: external-providers/golang-external-provider/pkg/golang/provider.go (hunk 3 above)
Comment (pranavgaikwad) [reply external-providers/golang-external-provider/pkg/golang/provider.go:56]: @fabianvf yeah, it looks like this: 
```
time="2023-07-17T14:50:07-04:00" level=error msg="unable to load rule set" error="stat /tmp/ruleset.yaml: no such file or directory" logger=parser
time="2023-07-17T14:50:07-04:00" level=error msg="unable to init the providers" error="invalid lspServerPath provided, unable to init java provider" provider=java
```

**Comment by pranavgaikwad** [comment]
Created https://github.com/konveyor/analyzer-lsp/issues/262 to follow up on refactoring labels

**Review by djzager — APPROVED** [review]

**Pull request merged**